consensus/core/src/authority_service.rs

```diff
@@ -315,15 +316,21 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
             .add_blocks(vec![verified_block.clone()])
             .await
             .map_err(|_| ConsensusError::Shutdown)?;
+
+        // Schedule fetching missing ancestors from this peer in the background.
         if !missing_ancestors.is_empty() {
-            // schedule the fetching of them from this peer
-            if let Err(err) = self
-                .synchronizer
-                .fetch_blocks(missing_ancestors, peer)
-                .await
-            {
-                warn!("Errored while trying to fetch missing ancestors via synchronizer: {err}");
-            }
+            self.context
+                .metrics
+                .node_metrics
+                .handler_received_block_missing_ancestors
+                .with_label_values(&[peer_hostname])
+                .inc_by(missing_ancestors.len() as u64);
+            let synchronizer = self.synchronizer.clone();
+            spawn_monitored_task!(async move {
+                if let Err(err) = synchronizer.fetch_blocks(missing_ancestors, peer).await {
+                    debug!("Failed to fetch missing ancestors via synchronizer: {err}");
```

Comment:
nit: maybe worth adding that missing ancestors will be attempted to fetch later which is why this is not necessarily an issue.

Reply:
It would be hard to explain it cleanly. The failed fetches will likely first be retried by other live fetches, then periodic sync. Keep this in the comment. This line becomes `debug` which will not be observed by operators usually.

Reply:
Are you using the `spawn_monitored_task` to make this cancel safe?